src/main/java/org/jnode/fs/xfs/XfsDirectory.java

```diff
@@ -33,13 +35,10 @@ public class XfsDirectory extends AbstractFSDirectory implements FSDirectoryId {
     /**
      * The related entry.
      */
+    @NonNull
+    @Getter
     private final XfsEntry entry;
 
-    /**
-     * The inode.
-     */
-    private final INode inode;
```

Comment:
No need to have this field, which only adds the complexity of reading, (because we use inode in some places, and use entry.getINode() in other places, which are exactly the same)
see line 58 in previous code, it is just `inode = entry.getINode()`